### ISSUE TITLE
Support for multiple registration periods.

### ISSUE THREAD

**Issue by Jambrose777** [thewca/worldcubeassociation.org#3267]
While working on #3257 , I realized that there are many competitions that utilize multiple registration periods. This is either done by an external website or by changing the start and end registration period in WCA. To be as accurate as possible it would be useful to have the registration periods be a list that you can add periods onto.

**Comment by jfly** [comment]
@Jambrose777, could you start a list of "things" we want to store about each registration period? I assume it's more than just a start date and an end date. Maybe there are different registration fees associated with each period? Other stuff?

**Comment by cubewhiz** [comment]
I've been making a list of things we should have or change from the registration requirements, too. Maybe it's best to just have one list with checkboxes that we can cross off as they are implemented?

**Comment by jfly** [comment]
Yes please, put together a single list, and we can break it up into discrete pieces of work later on.

**Comment by AlbertoPdRF** [comment]
This request is also included in #3526, maybe this issue can be closed?

**Comment by Jambrose777** [comment]
Some technical comments for this issue:
- [ ] Convert the following fields from the Competitions table to their own registration_periods table: registration_open, registration_close, base_entry_fee_lowest_denomination, currency_code (though currency_code may be global)
- [ ] Convert all existing registration periods to the new table
- [ ] Allow multiple registration periods to be added via the edit Competitions page
- [ ] Add anything relevant from #3526

Eventually we would also link this with Multiple Venues in some capacity, but the two should certainly be implemented independently. #5988
